File: src/Entity/Partenaire.php
<?php

namespace App\Entity;

use App\Repository\PartenaireRepository;
use Doctrine\ORM\Mapping as ORM;
use App\Entity\Users;

#[ORM\Entity(repositoryClass: PartenaireRepository::class)]
#[ORM\Table(name: 'partenaire')]
class Partenaire
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: Users::class)]
    #[ORM\JoinColumn(nullable: true)]
    private ?Users $user = null;

    #[ORM\Column(length: 200)]
    private string $nomEntreprise = '';

    #[ORM\Column(nullable: true, length: 50)]
    private ?string $ice = null;

    #[ORM\Column(nullable: true, length: 100)]
    private ?string $responsableNom = null;

    #[ORM\Column(nullable: true, length: 20)]
    private ?string $responsableTelephone = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $adresseEntreprise = null;

    #[ORM\Column(nullable: true, length: 200)]
    private ?string $siteWeb = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $description = null;

    #[ORM\Column(nullable: true, length: 50)]
    private ?string $statut = null;

    #[ORM\Column(type: 'date', nullable: true)]
    private ?\DateTimeInterface $dateValidation = null;

    #[ORM\Column(type: 'decimal', nullable: true, precision: 5, scale: 2)]
    private ?string $commission = null;

    #[ORM\Column(type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $dateInscription = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getUser(): ?Users
    {
        return $this->user;
    }

    public function setUser(?Users $user): static
    {
        $this->user = $user;
        return $this;
    }

    public function getNomEntreprise(): string
    {
        return $this->nomEntreprise;
    }

    public function setNomEntreprise(string $nomEntreprise): static
    {
        $this->nomEntreprise = $nomEntreprise;
        return $this;
    }

    public function getIce(): ?string
    {
        return $this->ice;
    }

    public function setIce(?string $ice): static
    {
        $this->ice = $ice;
        return $this;
    }

    public function getResponsableNom(): ?string
    {
        return $this->responsableNom;
    }

    public function setResponsableNom(?string $responsableNom): static
    {
        $this->responsableNom = $responsableNom;
        return $this;
    }

    public function getResponsableTelephone(): ?string
    {
        return $this->responsableTelephone;
    }

    public function setResponsableTelephone(?string $responsableTelephone): static
    {
        $this->responsableTelephone = $responsableTelephone;
        return $this;
    }

    public function getAdresseEntreprise(): ?string
    {
        return $this->adresseEntreprise;
    }

    public function setAdresseEntreprise(?string $adresseEntreprise): static
    {
        $this->adresseEntreprise = $adresseEntreprise;
        return $this;
    }

    public function getSiteWeb(): ?string
    {
        return $this->siteWeb;
    }

    public function setSiteWeb(?string $siteWeb): static
    {
        $this->siteWeb = $siteWeb;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): static
    {
        $this->description = $description;
        return $this;
    }

    public function getStatut(): ?string
    {
        return $this->statut;
    }

    public function setStatut(?string $statut): static
    {
        $this->statut = $statut;
        return $this;
    }

    public function getDateValidation(): ?\DateTimeInterface
    {
        return $this->dateValidation;
    }

    public function setDateValidation(?\DateTimeInterface $dateValidation): static
    {
        $this->dateValidation = $dateValidation;
        return $this;
    }

    public function getCommission(): ?float
    {
        return $this->commission !== null ? (float)$this->commission : null;
    }

    public function setCommission(?float $commission): static
    {
        $this->commission = $commission !== null ? (string)$commission : null;
        return $this;
    }

    public function getDateInscription(): ?\DateTimeInterface
    {
        return $this->dateInscription;
    }

    public function setDateInscription(?\DateTimeInterface $dateInscription): static
    {
        $this->dateInscription = $dateInscription;
        return $this;
    }

}

File: nexora_java/src/main/java/com/pi/entities/Partenaire.java
package com.pi.entities;

import java.util.Date;

public class Partenaire {
    private int id;
    private int userId;
    private String nomEntreprise;
    private String ice;
    private String responsableNom;
    private String responsableTelephone;
    private String adresseEntreprise;
    private String siteWeb;
    private String description;
    private String statut;
    private Date dateValidation;
    private double commission;
    private Date dateInscription;

    // Constructeur vide
    public Partenaire() {}

    // Constructeur avec paramètres principaux
    public Partenaire(int userId, String nomEntreprise, String responsableNom) {
        this.userId = userId;
        this.nomEntreprise = nomEntreprise;
        this.responsableNom = responsableNom;
        this.statut = "actif";
        this.dateInscription = new Date();
        this.commission = 10.0;
    }

    // Getters et Setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public int getUserId() { return userId; }
    public void setUserId(int userId) { this.userId = userId; }

    public String getNomEntreprise() { return nomEntreprise; }
    public void setNomEntreprise(String nomEntreprise) { this.nomEntreprise = nomEntreprise; }

    public String getIce() { return ice; }
    public void setIce(String ice) { this.ice = ice; }

    public String getResponsableNom() { return responsableNom; }
    public void setResponsableNom(String responsableNom) { this.responsableNom = responsableNom; }

    public String getResponsableTelephone() { return responsableTelephone; }
    public void setResponsableTelephone(String responsableTelephone) { this.responsableTelephone = responsableTelephone; }

    public String getAdresseEntreprise() { return adresseEntreprise; }
    public void setAdresseEntreprise(String adresseEntreprise) { this.adresseEntreprise = adresseEntreprise; }

    public String getSiteWeb() { return siteWeb; }
    public void setSiteWeb(String siteWeb) { this.siteWeb = siteWeb; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public String getStatut() { return statut; }
    public void setStatut(String statut) { this.statut = statut; }

    public Date getDateValidation() { return dateValidation; }
    public void setDateValidation(Date dateValidation) { this.dateValidation = dateValidation; }

    public double getCommission() { return commission; }
    public void setCommission(double commission) { this.commission = commission; }

    public Date getDateInscription() { return dateInscription; }
    public void setDateInscription(Date dateInscription) { this.dateInscription = dateInscription; }

    @Override
    public String toString() {
        return "Partenaire #" + id + " - " + nomEntreprise + " (" + responsableNom + ")";
    }
}